R/utils-format.r

```diff
@@ -277,6 +277,21 @@ tibble_glimpse_width <- function(width) {
   getOption("width")
 }
 
+pluralise_msg <- function(message, objects) {
+  paste0(pluralise(message, objects), format_n(objects))
+}
+
+pluralise <- function(message, objects) {
+  stopifnot(length(objects) > 0)
+  if (length(objects) == 1) {
+    # strip [, remove everything within ()
+    gsub("\\[|\\]|\\([^\\)]+\\)", "", message)
```

Comment:
Can we match only pairs of opening and closing parens/brackets that don't have spaces inside, e.g.

```
"[[]([^] ]+)]" -> "\\1"
"[(]([^) ]+)[)]" -> ""
```

Example:

```
> gsub("[[]([^] ]+)]", "\\1", "[an] index(es)")
[1] "an index(es)"
> gsub("[(]([^) ]+)[)]", "", "[an] index(es)")
[1] "[an] index"
```